src/rgw/driver/rados/rgw_data_sync.cc

```diff
@@ -1109,21 +1110,28 @@ class RGWDataSyncShardMarkerTrack : public RGWSyncShardMarkerTrack<string, strin
   rgw_data_sync_marker sync_marker;
   RGWSyncTraceNodeRef tn;
   RGWObjVersionTracker& objv;
+  sync_deltas::SyncDeltaCountersManager sync_delta_counters_manager;
 
 public:
   RGWDataSyncShardMarkerTrack(RGWDataSyncCtx *_sc,
+                         const uint32_t shard_id,
                          const string& _marker_oid,
                          const rgw_data_sync_marker& _marker,
                          RGWSyncTraceNodeRef& _tn, RGWObjVersionTracker& objv) : RGWSyncShardMarkerTrack(DATA_SYNC_UPDATE_MARKER_WINDOW),
                                                                 sc(_sc), sync_env(_sc->env),
                                                                 marker_oid(_marker_oid),
                                                                 sync_marker(_marker),
-                                                                tn(_tn), objv(objv) {}
+                                                                tn(_tn), objv(objv),
+                                                                sync_delta_counters_manager(ceph::perf_counters::key_create(rgw_sync_delta_counters_key, {{"source-zone", sync_env->svc->zone->get_zone_params().get_id()}, {"dest-zone", _sc->source_zone.id}, {"shard-id", std::to_string(shard_id)}}), _sc->env->cct) {}
```

Comment:
i think you have the source-zone and dest-zone values mixed up. `svc->zone->get_zone_params()` corresponds to the local (dest) zone